pom.xml

```
@@ -130,5 +130,11 @@
             <version>1.0.5</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.jenkins</groupId>
+            <artifactId>configuration-as-code</artifactId>
+            <version>1.0</version>
```

use a newer version of configuration-as-code please,
```suggestion
            <version>1.4</version>
```

---

Thanks for the reviewing ! But now i cannot proceed because of the test failure (and thats  also why i left the test case just a skeleton). Can you take a look at the error message since I have no idea what went wrong there in the latest commit. (Will it be the version )

brief descirption of the error message : unclassified: error configuring 'unclassified' with class io.jenkins.plugins.casc.impl.configurators.GlobalConfigurationCategoryConfigurator configurator

gist : https://gist.github.com/imaffe/cb42e96f75e11383f24a156e82889364

Thanks !

---

Its trying to call the `setDiskPools` method but the list is an `AbstractList` and its throwing an `UnsupportedOperationException` 

If you try what I suggested here it might work:
https://github.com/jenkinsci/external-workspace-manager-plugin/pull/56#discussion_r251303959

---

Thanks very much ! I'll take a look at these stuff when the class is over . Thanks very much .

---

Wow, I learned something here. Thank you @timja.

I re-read the List documentation and some of its methods can throw the `UnsupportedOperationException`, including the [`clear()`](https://docs.oracle.com/javase/8/docs/api/java/util/List.html#clear--) method.

Here is my understanding: Since in this case the `List` is a `List` and not an extension of `List`, the `clear()` method is [optional](https://docs.oracle.com/javase/8/docs/api/java/util/List.html#clear--) and does not have an implementation, it can throw the exception instead. If the `List` was `ArrayList` (as an example), calling `clear()` would work since `clear()` is [not optional](https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#clear--) for `ArrayList`, and it has to implement the method.

Under [Collection Framework Overview](https://docs.oracle.com/javase/8/docs/technotes/guides/collections/overview.html), you can find a discussion about this topic.